common/reducers/swap.js

```diff
@@ -0,0 +1,60 @@
+import {
+    SWAP_DESTINATION_AMOUNT,
+    SWAP_DESTINATION_KIND,
+    SWAP_ORIGIN_AMOUNT,
+    SWAP_ORIGIN_KIND,
+    SWAP_UPDATE_BITY_RATES
+} from 'actions/swap';
+
+import {without} from 'lodash';
+const ALL_CRYPTO_KIND_OPTIONS = ['BTC', 'ETH', 'REP'];
+
+const initialState = {
+    originAmount: 0,
+    destinationAmount: 0,
+    originKind: 'BTC',
+    destinationKind: 'ETH',
+    destinationKindOptions: without(ALL_CRYPTO_KIND_OPTIONS, 'BTC'),
+    originKindOptions: without(ALL_CRYPTO_KIND_OPTIONS, 'REP'),
+    bityRates: {}
+};
+
+
+export function swap(state = initialState, action) {
+    switch (action.type) {
+        case SWAP_ORIGIN_KIND: {
+            return {
+                ...state,
+                originKind: action.value,
+                destinationKindOptions: without(ALL_CRYPTO_KIND_OPTIONS, action.value),
+                destinationKind: without(ALL_CRYPTO_KIND_OPTIONS, action.value)[0]
```

Comment:
yeah, there is. I could write up a comment explaining what the default state is, it matches the existing v3.